autosklearn/ensembles/ensemble_selection.py

```diff
@@ -27,6 +27,17 @@ def __init__(
         self.mode = mode
         self.random_state = random_state
 
+    def __getstate__(self) -> Dict[str, Any]:
+        # Cannot serialize a metric if
+        # it is user defined.
+        # That is, if doing pickle dump
+        # the metric won't be the same as the
+        # one in __main__. we don't use the metric
+        # in the EnsembleSelection so this should
+        # be fine
```

Comment:
I disagree on this - we use `self.metric` to compute the scores. Does this issue happen in the examples on custom metrics?

Reply:
We discussed offline:

- We can proceed with removing the metrics from the dump pickle object. Examples work and unit test is not failing. 
- We will update the ensemble selection fit method to print a better message in case metric is None.